articlequality/feature_lists/wikidatawiki.py

```diff
@@ -62,6 +63,16 @@ def _process_external_identifiers(entity):
     _process_external_identifiers,
     depends_on=[wikibase_.revision.datasources.entity])
 
+def _process_commons_media(entity):
+    for pid, statements in entity.properties.items():
+        if pid in property_datatypes.COMMONS_MEDIA: # FIXME: COMMONS_MEDIA doesn't exist yet
+            return True
```

Comment:
I think it would be better to search for the pid directly.  Right now, this has linear complexity as it has to scan every PID in the entity in order to evaluate.  

If you can make COMMONS_MEDIA into a set, you can do something like this:
```
return len(entity.properties.keys() & property_datatypes.COMMONS_MEDIA) > 0
```

This performs a set intersection and will return true if there is any overlap.

Reply:
Makes sense to me, though I might do it in an extra pull request focussed on performance.

Reply:
Yes, I'd say let's benchmark it and run profilers. In the profilers I ran with ores, this didn't come up as a bottleneck (the reversing data source dependency graph took a lot instead, fixing that would gain us a lot)

Reply:
What do you mean by "reversing data source dependency graph" and what might be fixed?

Generally, using the right data structure/indexing method is good practice regardless if it shows up at the top of the profiler or not.  Little inefficiencies like this stack up.   Here you have an O(n) problem with an O(n^2) solution.  You're cycling through the commons media list (should be set()) for *every property in the item*.  So if you have an item with 50 properties and a list with 50 properties, you're going to do 2500 operations when you could just do 50.   Later these collections of properties might get even bigger and the problem will get worse in surprising ways to future developers. 

What I proposed here is not premature optimization.  This is using the right data structure for the job. 

Here's a simple demonstration: https://gist.github.com/halfak/f6e18823eaf1a40fd0868ff17a2a0688